app/uk/gov/hmrc/economiccrimelevyregistration/controllers/NotableErrorController.scala

```diff
@@ -92,4 +94,18 @@ class NotableErrorController @Inject() (
     Ok(partyTypeMismatchView())
   }
 
+  def verificationFailed: Action[AnyContent] = detailsDoNotMatch
+
+  def detailsDoNotMatch: Action[AnyContent] = (authoriseWithoutEnrolmentCheck andThen getRegistrationData) {
+    implicit request =>
+      request.registration.entityType match {
+        case Some(entityType) =>
+          entityType match {
+            case UkLimitedCompany => Ok(detailsDoNotMatchView("ct"))
+            case _                => Ok(detailsDoNotMatchView("sa"))
```

Comment:
Thinking rather than have a single view - have a view for SA and a view for CT, it will be some duplication but will avoid you needing to do this and then doesn't couple the structures of the views